service/storage_proxy.cc

```diff
@@ -1245,7 +1245,12 @@ future<> paxos_response_handler::learn_decision(paxos::proposal decision, bool a
                 auto meta = std::move(std::get<1>(t));
                 // Pick only the CDC ("augmenting") mutations
                 auto range = boost::make_iterator_range(mutations.end() - meta.total_cdc_mutations, mutations.end());
-                return _proxy->mutate_internal(std::move(range), _cl_for_learn, false, tr_state, _permit, _timeout);
+                // Need to adjust ranges in the metadata, because we are going to skip non-CDC mutations
+                for (auto& range : meta.generated_ranges) {
```

Comment:
Please don't use the variable name "range" here, it's very confusing because you have a different variable "range" just one line up!

Reply:
No longer relevant to the newest version